boa/src/builtins/value/display.rs

```diff
@@ -1,5 +1,11 @@
 use super::*;
 
+/// This object is used for displaying a `Value`.
+#[derive(Debug, Clone, Copy)]
+pub struct ValueDisplay<'value> {
+    pub(super) value: &'value Value,
```

Comment:
I like this idea. In the future, this will also need a `ctx` or something with an interner, probably.

Why does it need to be `pub(super)`?

Reply:
> Why does it need to be `pub(super)`?

Its only exposed to `Value`.

> In the future, this will also need a `ctx` or something with an interner, probably.

why would this need `ctx` this displays the `Value` without context for `Display` trait, also why interner>

Reply:
In the future we could add a `.display_with_settings()`
```rust
struct DispplaySettings {
	use_double_quote: bool,
	indent: String,
}
```
that would the Value with different setting such as `''` instead of `"` use different indent (or specify a tab), etc.

Reply:
> > Why does it need to be `pub(super)`?
> 
> Its only exposed to `Value`.

Is it used?

> > In the future, this will also need a `ctx` or something with an interner, probably.
> 
> why would this need `ctx`, this displays the `Value` without context for `Display` trait, also why interner?

If we start interning immutable strings, we might use them in `Value`, right? At least we should use them in `TokenKind` for sure. So when displaying them, we might need the interner to resolve the indexes.

Reply:
> Is it used?

Yes, since it is defined in `value/display.rs` we can't do `ValueDispaly { value: self }` in `value/mod.rs`

> If we start interning immutable strings, we might use them in `Value`, right? At least we should use them in `TokenKind` for sure. So when displaying them, we might need the interner to resolve the indexes.

Yes. the interner would be used for `TokenKind` but only for identifiers, and `Value` would not know what a interned string is. If we use it for stings with have the same issue: https://github.com/boa-dev/boa/issues/279#issuecomment-621945984

Reply:
> > Is it used?
> 
> Yes, since it is defined in `value/display.rs` we can't do `ValueDispaly { value: self }` in `value/mod.rs`

Hmmm not sure if we should instead be using a `new()` function in `ValueDisplay` if we only need it public for creation.

The PR can be merged as is, see if it makes sense to change that, but from my side, it's good :)

Reply:
> Hmmm not sure if we should instead be using a `new()` function in `ValueDisplay` if we only need it public for creation.

Since the only way `ValueDisplay` can be created is with `.display()` and we don't construct it anywere and it's not public for construction, I not see the reason why it would need a `new` constructor. the standard library does the same.